diff -Nur silverstripe-official/sapphire/core/model/Versioned.php silverstripe-tmp/sapphire/core/model/Versioned.php
--- silverstripe-official/sapphire/core/model/Versioned.php	2010-09-20 03:16:52.000000000 +0200
+++ silverstripe-tmp/sapphire/core/model/Versioned.php	2010-11-12 14:00:39.937020327 +0100
@@ -1023,6 +1023,10 @@
 	function cacheKeyComponent() {
 		return 'stage-'.self::current_stage();
 	}
+	
+	function getStages() {
+		return $this->stages;
+	}
 }
 
 /**
